sycl/source/detail/graph_impl.hpp

```diff
@@ -48,8 +83,13 @@ class node_impl {
   std::vector<std::weak_ptr<node_impl>> MPredecessors;
   /// Type of the command-group for the node.
   sycl::detail::CG::CGTYPE MCGType = sycl::detail::CG::None;
+  /// User facing type of the node
+  node_type MNodeType = node_type::empty;
   /// Command group object which stores all args etc needed to enqueue the node
   std::unique_ptr<sycl::detail::CG> MCommandGroup;
+  /// Stores the executable graph impl associated with this node if it is a
+  /// subgraph node.
+  std::shared_ptr<exec_graph_impl> MSubGraphImpl;
```

Comment:
I don't believe so, the subgraph node is unique to the parent modifiable graph, and is not referenced anywhere in `MSubGraphImpl`. I think it is important to extend the lifetime of the subgraph impl here, however we don't specify that adding a subgraph node extends the lifetime of the executable graph in the spec. I think it should be valid, though I am not sure whether this would require additional spec language or if it's just an implementation detail.

Reply:
Thanks for explaining 👍 This is too much of an implementation detail for spec language I think.